src/Agent.Listener/CommandSettings.cs

```diff
@@ -114,8 +115,14 @@ public CommandSettings(IHostContext context, string[] args)
                 secretArgNames: Constants.Agent.CommandLine.Args.Secrets);
             _parser.Parse(args);
 
+            if (environmentScope == null)
+            {
+                environmentScope = new SystemEnvironment();
+            }
+
             // Store and remove any args passed via environment variables.
-            IDictionary environment = Environment.GetEnvironmentVariables();
+            var environment = environmentScope.GetEnvironmentVariables();
```

Comment:
Doesn't this approach mean we're no longer testing SystemEnvironments (or at least testing less robustly), and by extension not testing WindowsEnvHelper? That seems scary to me, especially given the changes we're making to WindowsEnvHelper (which _seem_ fine, but I don't really fully understand)

Reply:
The WindowsEnvHelper is a separate change is orthogonal to the changes to CommandSettings.

Reply:
Oh whoops, I got my wires crossed - nevermind